test/Npgsql.Tests/SystemTransactionTests.cs

```diff
@@ -150,17 +153,14 @@ public void Close_connection()
             scope.Complete();
         }
         AssertNumberOfRows(1);
-        Assert.True(PoolManager.Pools.TryGetValue(connString, out var pool));
-        Assert.That(pool!.Statistics.Idle, Is.EqualTo(1));
-
-        using (var conn = new NpgsqlConnection(connString))
-            NpgsqlConnection.ClearPool(conn);
+        Assert.That(dataSource.Statistics.Idle, Is.EqualTo(1));
     }
 
     [Test]
     public void Enlist_to_two_transactions()
     {
-        using var conn = OpenConnection(ConnectionStringEnlistOff);
+        using var dataSource = CreateDataSource(ConnectionStringEnlistOff);
```

Comment:
As in DistributedTransactionTests, we should be able to share the data source across tests